modular_zubbers/code/datums/components/vore/belly.dm

```diff
@@ -0,0 +1,522 @@
+		if(!listener_vore_prefs)
+			continue
+		var/pref_enabled = listener_vore_prefs.read_preference(/datum/vore_pref/toggle/digestion_noises)
+		//We don't bother executing any other code if the prey doesn't want to hear the noises.
+		if(!pref_enabled)
+			// In case someone turns off their pref
+			listening_mob.stop_sound_channel(CHANNEL_PREYLOOP)
+			continue
+
+		// We don't want the sounds to overlap, but we do want them to steadily replay.
+		// We also don't want the sounds to play if the pred hasn't marked this belly as fleshy, or doesn't
+		// have the right sounds to play.
+		if(TIMER_COOLDOWN_FINISHED(listening_mob, COOLDOWN_PREYLOOP))
+			listening_mob.stop_sound_channel(CHANNEL_PREYLOOP)
+			// Must reload each time because playsound_local modifies the sound datum
+			var/sound/preyloop = sound('modular_zubbers/sound/vore/sunesound/prey/loop.ogg')
+			listening_mob.playsound_local(get_turf(src), preyloop, PREYLOOP_VOLUME, 0, channel = CHANNEL_PREYLOOP)
+			TIMER_COOLDOWN_START(listening_mob, COOLDOWN_PREYLOOP, 52 SECONDS)
+
+/// Handles prey entering a belly, and starts deep_search_prey
+/obj/vore_belly/Entered(atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+	. = ..()
+	if(!owner)
+		return
+	var/mob/living/living_parent = owner.parent
+	owner.play_vore_sound(get_insert_sound())
+	to_chat(living_parent, span_notice("[arrived] slides into your [lowertext(name)]."))
+	owner.appearance_holder.vis_contents += arrived
+	if(ismob(arrived))
+		var/mob/M = arrived
+		RegisterSignal(M, COMSIG_MOVABLE_USING_RADIO, PROC_REF(try_deny_radio))
+		ADD_TRAIT(M, TRAIT_SOFTSPOKEN, TRAIT_SOURCE_VORE)
+		deep_search_prey(M)
+		to_chat(M, examine_block("You slide into [span_notice("[owner.parent]")]'s [span_green(lowertext(name))]!\n[format_message(desc, M)]"))
+		// Add the appearance_holder to prey so they can see fellow prey
+		if(can_taste && iscarbon(M))
+			var/mob/living/carbon/H = M
+			to_chat(living_parent, span_notice("[H] tastes of [H.dna.features["taste"] || "nothing"]."))
+		if(M.client)
+			M.client.screen += owner.appearance_holder
+
+/obj/vore_belly/proc/try_deny_radio()
+	if(muffles_radio)
+		return COMPONENT_CANNOT_USE_RADIO
+	return null
+
+/// Search through prey's recursive contents to prevent smuggling any GLOB.vore_blacklist_types items around
+/obj/vore_belly/proc/deep_search_prey(mob/arrived)
+	var/turf/reject_location = get_turf(src)
+	var/list/all_contents = arrived.get_contents()
+	for(var/atom/movable/AM as anything in all_contents)
+		if(is_type_in_list(AM, GLOB.vore_blacklist_types))
+			// If it's directly in their inventory, call dropItemToGround so that it cleans up the hud
+			if(AM in arrived)
+				arrived.dropItemToGround(AM, TRUE)
+			AM.forceMove(reject_location)
+		#if DISABLES_SENSORS
+		if(istype(AM, /obj/item/clothing/under))
+			var/obj/item/clothing/under/sensor_clothing = AM
+			sensor_clothing.sensor_mode = NO_SENSORS
```

Comment:
This irreversibly disables sensors on a jumpsuit, even if someone has been ejected.

Reply:
it shouldn't? this is the var that is toggleable

Reply:
ah, I see. the define for a suit having NO_SENSORS was used, instead of `SENSOR_OFF`. different answer, same result.
Though, should still change it to `SENSOR_OFF `

https://github.com/Bubberstation/Bubberstation/blob/master/code/__DEFINES/clothing.dm#L9-L11

Reply:
Fix'd